docs/design/incrementaldownload.md

```diff
@@ -0,0 +1,65 @@
+# Incremental Download
+
+## Overview
+The feature is to allow Incremental Download of the Artifact in release so that all of the artifact items are not downloaded from the source provider and can be retrieved from a pre-maintained cache as well.
+
+## Task Changes
+Ideally we should be given the hashes of the files from the service itself but since services like Jenkins and Teamcity do not provide such hashes, we needed some generic way to get the hashes.
+
+A new task will be created named `Generate Artifact Metadata` that will generate `artifact-metadata.csv` file containing the filepaths and hashes of artifact items mapped to one another, created in the same directory as that of the artifacts.
+
+Currently, it supports VSTS but similar tasks can be written for Jenkins and teamcity as well. 
```

Comment:
This task can be used when build is generated with VSTS Build. Similar task can be written when build is generated by Jenkins/TeamCity etc.